[AST] Hit assertion failure when diagnosing argument label

**Describe the bug**
The compiler crashes while running the following reproducer. I have narrowed it down to remove the dependencies on Metal Performance Shaders Graph and Swift for TensorFlow.

```swift
struct MPSGraphTensor { }

struct MPSGraph {
  func addition(
    _ primaryTensor: MPSGraphTensor,
    _ secondaryTensor: MPSGraphTensor,
    name: String?
  ) {
    
  }
}

struct Tensor<T> {}

struct _ExecutionContext {
  func performBinaryOp<T>(
    _ lhs: Tensor<T>,
    _ rhs: Tensor<T>,
    _ op: (MPSGraphTensor, MPSGraphTensor, String?) -> MPSGraphTensor
  ) -> Tensor<T> {
    fatalError()
  }
}

public enum _RawTFEager {
  public static func addV2<T>(
    _ x: Tensor<T>
  ) -> Tensor<T> {
    return _ExecutionContext.performBinaryOp(x, x, MPSGraph.addition)
  }
}
```

**Error summary**
```
Assertion failed: (numMissing > 0 || numExtra > 0 || numWrong > 0), function diagnoseArgumentLabelError, file MiscDiagnostics.cpp, line 1884.
3.	While evaluating request TypeCheckSourceFileRequest(source_file "file.swift")
4.	While evaluating request TypeCheckFunctionBodyRequest(file.(file)._RawTFEager.addV2@file.swift:26:22)
5.	While type-checking statement at [file.swift:28:18 - line:30:3] RangeText="{
    return _ExecutionContext.performBinaryOp(x, x, MPSGraph.addition)
  "
6.	While type-checking statement at [file.swift:29:5 - line:29:69] RangeText="return _ExecutionContext.performBinaryOp(x, x, MPSGraph.addition"
7.	While type-checking expression at [file.swift:29:12 - line:29:69] RangeText="_ExecutionContext.performBinaryOp(x, x, MPSGraph.addition"
```

<details>
<summary>Full error</summary>

```
(base) philipturner@M1-Max-MacBook-Pro Desktop % swift file.swift
file.swift:26:22: error: method cannot be declared public because its parameter uses an internal type
  public static func addV2<T>(
                     ^
file.swift:13:8: note: type declared here
struct Tensor<T> {}
       ^
Assertion failed: (numMissing > 0 || numExtra > 0 || numWrong > 0), function diagnoseArgumentLabelError, file MiscDiagnostics.cpp, line 1884.
Stack dump:
0.	Program arguments: /Library/Developer/Toolchains/swift-DEVELOPMENT-SNAPSHOT-2022-05-18-a.xctoolchain/usr/bin/swift-frontend -frontend -interpret file.swift -Xllvm -aarch64-use-tbi -enable-objc-interop -sdk /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX12.3.sdk -color-diagnostics -new-driver-path /Library/Developer/Toolchains/swift-DEVELOPMENT-SNAPSHOT-2022-05-18-a.xctoolchain/usr/bin/swift-driver -empty-abi-descriptor -resource-dir /Library/Developer/Toolchains/swift-DEVELOPMENT-SNAPSHOT-2022-05-18-a.xctoolchain/usr/lib/swift -module-name file -target-sdk-version 12.3
1.	Apple Swift version 5.8-dev (LLVM 515f8ef5b9ec986, Swift 7435c936a6a67d0)
2.	Compiling with the current language version
3.	While evaluating request TypeCheckSourceFileRequest(source_file "file.swift")
4.	While evaluating request TypeCheckFunctionBodyRequest(file.(file)._RawTFEager.addV2@file.swift:26:22)
5.	While type-checking statement at [file.swift:28:18 - line:30:3] RangeText="{
    return _ExecutionContext.performBinaryOp(x, x, MPSGraph.addition)
  "
6.	While type-checking statement at [file.swift:29:5 - line:29:69] RangeText="return _ExecutionContext.performBinaryOp(x, x, MPSGraph.addition"
7.	While type-checking expression at [file.swift:29:12 - line:29:69] RangeText="_ExecutionContext.performBinaryOp(x, x, MPSGraph.addition"
Stack dump without symbol names (ensure you have llvm-symbolizer in your PATH or set the environment var `LLVM_SYMBOLIZER_PATH` to point to it):
0  swift-frontend           0x000000010509dd30 llvm::sys::PrintStackTrace(llvm::raw_ostream&, int) + 56
1  swift-frontend           0x000000010509cf90 llvm::sys::RunSignalHandlers() + 128
2  swift-frontend           0x000000010509e394 SignalHandler(int) + 304
3  libsystem_platform.dylib 0x000000019923f4a4 _sigtramp + 56
4  libsystem_pthread.dylib  0x0000000199227ee0 pthread_kill + 288
5  libsystem_c.dylib        0x0000000199162340 abort + 168
6  libsystem_c.dylib        0x0000000199161754 err + 0
7  swift-frontend           0x00000001053de9d8 swift::diagnoseArgumentLabelError(swift::ASTContext&, swift::ArgumentList const*, llvm::ArrayRef<swift::Identifier>, bool, swift::InFlightDiagnostic*) (.cold.2) + 0
8  swift-frontend           0x00000001017a2488 swift::diagnoseArgumentLabelError(swift::ASTContext&, swift::ArgumentList const*, llvm::ArrayRef<swift::Identifier>, bool, swift::InFlightDiagnostic*) + 2820
9  swift-frontend           0x00000001016f878c swift::constraints::RelabelArguments::diagnose(swift::constraints::Solution const&, bool) const + 68
10 swift-frontend           0x0000000101653c88 swift::constraints::ConstraintSystem::applySolutionFixes(swift::constraints::Solution const&) + 1528
11 swift-frontend           0x00000001016543d8 swift::constraints::ConstraintSystem::applySolution(swift::constraints::Solution&, swift::constraints::SolutionApplicationTarget) + 92
12 swift-frontend           0x0000000101819a18 swift::TypeChecker::typeCheckExpression(swift::constraints::SolutionApplicationTarget&, swift::OptionSet<swift::TypeCheckExprFlags, unsigned int>) + 568
13 swift-frontend           0x0000000101819758 swift::TypeChecker::typeCheckExpression(swift::Expr*&, swift::DeclContext*, swift::constraints::ContextualTypeInfo, swift::OptionSet<swift::TypeCheckExprFlags, unsigned int>) + 84
14 swift-frontend           0x00000001018d36c0 swift::ASTVisitor<(anonymous namespace)::StmtChecker, void, swift::Stmt*, void, void, void, void>::visit(swift::Stmt*) + 2964
15 swift-frontend           0x00000001018d2af8 bool (anonymous namespace)::StmtChecker::typeCheckStmt<swift::Stmt>(swift::Stmt*&) + 136
16 swift-frontend           0x00000001018d0a50 (anonymous namespace)::StmtChecker::typeCheckASTNode(swift::ASTNode&) + 140
17 swift-frontend           0x00000001018d2c6c swift::ASTVisitor<(anonymous namespace)::StmtChecker, void, swift::Stmt*, void, void, void, void>::visit(swift::Stmt*) + 320
18 swift-frontend           0x00000001018d1e70 bool (anonymous namespace)::StmtChecker::typeCheckStmt<swift::BraceStmt>(swift::BraceStmt*&) + 136
19 swift-frontend           0x00000001018d153c swift::TypeCheckFunctionBodyRequest::evaluate(swift::Evaluator&, swift::AbstractFunctionDecl*) const + 756
20 swift-frontend           0x0000000101ca3fc4 llvm::Expected<swift::TypeCheckFunctionBodyRequest::OutputType> swift::Evaluator::getResultUncached<swift::TypeCheckFunctionBodyRequest>(swift::TypeCheckFunctionBodyRequest const&) + 404
21 swift-frontend           0x0000000101ca3d48 llvm::Expected<swift::TypeCheckFunctionBodyRequest::OutputType> swift::Evaluator::getResultCached<swift::TypeCheckFunctionBodyRequest, (void*)0>(swift::TypeCheckFunctionBodyRequest const&) + 140
22 swift-frontend           0x0000000101c030c4 swift::TypeCheckFunctionBodyRequest::OutputType swift::evaluateOrDefault<swift::TypeCheckFunctionBodyRequest>(swift::Evaluator&, swift::TypeCheckFunctionBodyRequest, swift::TypeCheckFunctionBodyRequest::OutputType) + 52
23 swift-frontend           0x00000001018fe8f4 swift::TypeCheckSourceFileRequest::evaluate(swift::Evaluator&, swift::SourceFile*) const + 308
24 swift-frontend           0x00000001019006f8 llvm::Expected<swift::TypeCheckSourceFileRequest::OutputType> swift::Evaluator::getResultUncached<swift::TypeCheckSourceFileRequest>(swift::TypeCheckSourceFileRequest const&) + 400
25 swift-frontend           0x0000000101900498 llvm::Expected<swift::TypeCheckSourceFileRequest::OutputType> swift::Evaluator::getResultCached<swift::TypeCheckSourceFileRequest, (void*)0>(swift::TypeCheckSourceFileRequest const&) + 124
26 swift-frontend           0x00000001018fe6b4 swift::TypeCheckSourceFileRequest::OutputType swift::evaluateOrDefault<swift::TypeCheckSourceFileRequest>(swift::Evaluator&, swift::TypeCheckSourceFileRequest, swift::TypeCheckSourceFileRequest::OutputType) + 44
27 swift-frontend           0x0000000100c4a6c4 bool llvm::function_ref<bool (swift::SourceFile&)>::callback_fn<swift::CompilerInstance::performSema()::$_7>(long, swift::SourceFile&) + 16
28 swift-frontend           0x0000000100c46bcc swift::CompilerInstance::forEachFileToTypeCheck(llvm::function_ref<bool (swift::SourceFile&)>) + 160
29 swift-frontend           0x0000000100c46b00 swift::CompilerInstance::performSema() + 76
30 swift-frontend           0x0000000100bf98fc withSemanticAnalysis(swift::CompilerInstance&, swift::FrontendObserver*, llvm::function_ref<bool (swift::CompilerInstance&)>, bool) + 60
31 swift-frontend           0x0000000100bed29c swift::performFrontend(llvm::ArrayRef<char const*>, char const*, void*, swift::FrontendObserver*) + 3392
32 swift-frontend           0x0000000100ae09e0 swift::mainEntry(int, char const**) + 3764
33 dyld                     0x000000010da7908c start + 520
zsh: abort      swift file.swift
```

</details>

**Environment (please complete the following information):**
 - OS: macOS 12.4
 - Toolchain Version: May 18, 2022 development snapshot


**Additional context**
This is interfering with development of my proof-of-concept Metal backend for [Swift for TensorFlow](https://github.com/s4tf/s4tf) (in a private repo). Although I can work around it at the moment, it would be nice to be able to call `MPSGraph` functions like this:
```swift
public static func addV2<T: TensorFlowNumeric>(
  _ x: Tensor<T>,
  _ y: Tensor<T>
) -> Tensor<T> {
  return _ExecutionContext.performBinaryOp(x, y, MPSGraph.addition)
}
```

Note that adding the correct argument labels still causes a crash:
```swift
public static func addV2<T: TensorFlowNumeric>(
  _ x: Tensor<T>,
  _ y: Tensor<T>
) -> Tensor<T> {
  return _ExecutionContext.performBinaryOp(x, y, MPSGraph.addition(_:_:name:))
}
```

While adding incorrect labels triggers an error diagnostic, avoiding the crash:
```swift
public static func addV2<T: TensorFlowNumeric>(
  _ x: Tensor<T>,
  _ y: Tensor<T>
) -> Tensor<T> {
  return _ExecutionContext.performBinaryOp(x, y, MPSGraph.addition(_:_:name:name:))
}
```
<details>
<summary>Correct, graceful early exit</summary>

```
(base) philipturner@M1-Max-MacBook-Pro Desktop % swift file.swift
file.swift:16:17: error: static methods may only be declared on a type
  public static func addV2<T>(
         ~~~~~~~^
         
file.swift:16:22: error: function cannot be declared public because its parameter uses an internal type
  public static func addV2<T>(
                     ^
file.swift:3:8: note: type declared here
struct Tensor<T> {}
       ^
file.swift:19:12: error: instance member 'performBinaryOp' cannot be used on type '_ExecutionContext'; did you mean to use a value of this type instead?
    return _ExecutionContext.performBinaryOp(x, x, MPSGraph.addition(_:_:name:name:))
           ^~~~~~~~~~~~~~~~~
file.swift:19:61: error: type 'MPSGraph' has no member 'addition(_:_:name:name:)'
    return _ExecutionContext.performBinaryOp(x, x, MPSGraph.addition(_:_:name:name:))
                                                   ~~~~~~~~ ^~~~~~~~~~~~~~~~~~~~~~~~
```

</details>

I bisected the origin of the bug to between the following dates. On the Swift 5.3 snapshot, it produces error diagnostics and doesn't crash. If you remove the `public` modifier, it won't remove all of the errors. It just says that the type is ambiguous.
- July 22, 2020 development snapshot
- January 27, 2021 development snapshot

<details>
<summary>Error log for 2020-07-22</summary>

```
(base) philipturner@m1-max-mbp Desktop % export TOOLCHAINS=org.swift.202007221a
(base) philipturner@m1-max-mbp Desktop % export TOOLCHAINS=org.swift.50202007221a
(base) philipturner@m1-max-mbp Desktop % swift --version
Apple Swift version 5.3-dev (LLVM e966de21c9ff9ed, Swift 493b4a8ef836af0)
Target: x86_64-apple-darwin21.5.0
(base) philipturner@m1-max-mbp Desktop % swift file.swift
file.swift:26:22: error: method cannot be declared public because its parameter uses an internal type
  public static func addV2<T>(
                     ^
file.swift:13:8: note: type declared here
struct Tensor<T> {}
       ^
file.swift:29:30: error: type of expression is ambiguous without more context
    return _ExecutionContext.performBinaryOp(x, x, MPSGraph.addition)
           ~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
(base) philipturner@m1-max-mbp Desktop % swift file.swift
file.swift:29:30: error: type of expression is ambiguous without more context
    return _ExecutionContext.performBinaryOp(x, x, MPSGraph.addition)
           ~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

</details>

<details>
<summary>Error log for 2021-01-27</summary>

```
(base) philipturner@m1-max-mbp Desktop % export TOOLCHAINS=org.swift.50202101271a
(base) philipturner@m1-max-mbp Desktop % swift --version
Apple Swift version 5.4-dev (LLVM e2976fe639d1f50, Swift f9be289b9e04b3d)
Target: arm64-apple-darwin21.5.0
(base) philipturner@m1-max-mbp Desktop % swift file.swift
file.swift:26:22: error: method cannot be declared public because its parameter uses an internal type
  public static func addV2<T>(
                     ^
file.swift:13:8: note: type declared here
struct Tensor<T> {}
       ^
Assertion failed: (numMissing > 0 || numExtra > 0 || numWrong > 0), function diagnoseArgumentLabelError, file /Users/buildnode/jenkins/workspace/oss-swift-package-macos/swift/lib/Sema/MiscDiagnostics.cpp, line 1834.
Please submit a bug report (https://swift.org/contributing/#reporting-bugs) and include the project and the crash backtrace.
Stack dump:
0.	Program arguments: /Library/Developer/Toolchains/swift-DEVELOPMENT-SNAPSHOT-2021-01-27-a.xctoolchain/usr/bin/swift-frontend -frontend -interpret file.swift -Xllvm -aarch64-use-tbi -enable-objc-interop -sdk /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX12.3.sdk -color-diagnostics -target-sdk-version 12.3 -module-name file 
1.	Apple Swift version 5.4-dev (LLVM e2976fe639d1f50, Swift f9be289b9e04b3d)
2.	While evaluating request TypeCheckSourceFileRequest(source_file "file.swift")
3.	While evaluating request TypeCheckFunctionBodyRequest(file.(file)._RawTFEager.addV2@file.swift:26:22)
4.	While type-checking statement at [file.swift:28:18 - line:30:3] RangeText="{
    return _ExecutionContext.performBinaryOp(x, x, MPSGraph.addition)
  "
5.	While type-checking statement at [file.swift:29:5 - line:29:69] RangeText="return _ExecutionContext.performBinaryOp(x, x, MPSGraph.addition"
6.	While type-checking expression at [file.swift:29:12 - line:29:69] RangeText="_ExecutionContext.performBinaryOp(x, x, MPSGraph.addition"
0  swift-frontend           0x00000001043d79fc llvm::sys::PrintStackTrace(llvm::raw_ostream&) + 52
1  swift-frontend           0x00000001043d6c5c llvm::sys::RunSignalHandlers() + 128
2  swift-frontend           0x00000001043d7fe4 SignalHandler(int) + 292
3  libsystem_platform.dylib 0x000000019923f4a4 _sigtramp + 56
4  libsystem_pthread.dylib  0x0000000199227ee0 pthread_kill + 288
5  libsystem_c.dylib        0x0000000199162340 abort + 168
6  libsystem_c.dylib        0x0000000199161754 err + 0
7  swift-frontend           0x0000000104732aa0 swift::diagnoseArgumentLabelError(swift::ASTContext&, swift::Expr*, llvm::ArrayRef<swift::Identifier>, bool, swift::InFlightDiagnostic*) (.cold.2) + 0
8  swift-frontend           0x0000000101271c44 swift::diagnoseArgumentLabelError(swift::ASTContext&, swift::Expr*, llvm::ArrayRef<swift::Identifier>, bool, swift::InFlightDiagnostic*) + 2696
9  swift-frontend           0x00000001011e90b4 swift::constraints::RelabelArguments::diagnose(swift::constraints::Solution const&, bool) const + 64
10 swift-frontend           0x000000010115f758 swift::constraints::ConstraintSystem::applySolutionFixes(swift::constraints::Solution const&) + 1540
11 swift-frontend           0x000000010115ff8c swift::constraints::ConstraintSystem::applySolution(swift::constraints::Solution&, swift::constraints::SolutionApplicationTarget) + 76
12 swift-frontend           0x00000001012ca804 swift::TypeChecker::typeCheckExpression(swift::constraints::SolutionApplicationTarget&, swift::OptionSet<swift::TypeCheckExprFlags, unsigned int>) + 620
13 swift-frontend           0x00000001012ca510 swift::TypeChecker::typeCheckExpression(swift::Expr*&, swift::DeclContext*, swift::constraints::ContextualTypeInfo, swift::OptionSet<swift::TypeCheckExprFlags, unsigned int>) + 80
14 swift-frontend           0x000000010136d218 swift::ASTVisitor<(anonymous namespace)::StmtChecker, void, swift::Stmt*, void, void, void, void>::visit(swift::Stmt*) + 3788
15 swift-frontend           0x000000010136c318 bool (anonymous namespace)::StmtChecker::typeCheckStmt<swift::Stmt>(swift::Stmt*&) + 136
16 swift-frontend           0x000000010136a6a4 (anonymous namespace)::StmtChecker::typeCheckASTNode(swift::ASTNode&) + 144
17 swift-frontend           0x000000010136c488 swift::ASTVisitor<(anonymous namespace)::StmtChecker, void, swift::Stmt*, void, void, void, void>::visit(swift::Stmt*) + 316
18 swift-frontend           0x000000010136b720 bool (anonymous namespace)::StmtChecker::typeCheckStmt<swift::BraceStmt>(swift::BraceStmt*&) + 136
19 swift-frontend           0x000000010136ae18 swift::TypeCheckFunctionBodyRequest::evaluate(swift::Evaluator&, swift::AbstractFunctionDecl*) const + 828
20 swift-frontend           0x0000000101640d08 llvm::Expected<swift::TypeCheckFunctionBodyRequest::OutputType> swift::Evaluator::getResultUncached<swift::TypeCheckFunctionBodyRequest>(swift::TypeCheckFunctionBodyRequest const&) + 400
21 swift-frontend           0x0000000101640a90 llvm::Expected<swift::TypeCheckFunctionBodyRequest::OutputType> swift::Evaluator::getResultCached<swift::TypeCheckFunctionBodyRequest, (void*)0>(swift::TypeCheckFunctionBodyRequest const&) + 140
22 swift-frontend           0x00000001015acd24 swift::TypeCheckFunctionBodyRequest::OutputType swift::evaluateOrDefault<swift::TypeCheckFunctionBodyRequest>(swift::Evaluator&, swift::TypeCheckFunctionBodyRequest, swift::TypeCheckFunctionBodyRequest::OutputType) + 52
23 swift-frontend           0x0000000101396cc4 swift::TypeCheckSourceFileRequest::evaluate(swift::Evaluator&, swift::SourceFile*) const + 312
24 swift-frontend           0x0000000101397c80 llvm::Expected<swift::TypeCheckSourceFileRequest::OutputType> swift::Evaluator::getResultUncached<swift::TypeCheckSourceFileRequest>(swift::TypeCheckSourceFileRequest const&) + 396
25 swift-frontend           0x0000000101397a24 llvm::Expected<swift::TypeCheckSourceFileRequest::OutputType> swift::Evaluator::getResultCached<swift::TypeCheckSourceFileRequest, (void*)0>(swift::TypeCheckSourceFileRequest const&) + 120
26 swift-frontend           0x0000000101396a94 swift::TypeCheckSourceFileRequest::OutputType swift::evaluateOrDefault<swift::TypeCheckSourceFileRequest>(swift::Evaluator&, swift::TypeCheckSourceFileRequest, swift::TypeCheckSourceFileRequest::OutputType) + 44
27 swift-frontend           0x0000000100813c8c swift::CompilerInstance::forEachFileToTypeCheck(llvm::function_ref<void (swift::SourceFile&)>) + 144
28 swift-frontend           0x0000000100813bd0 swift::CompilerInstance::performSema() + 76
29 swift-frontend           0x000000010071f384 withSemanticAnalysis(swift::CompilerInstance&, swift::FrontendObserver*, llvm::function_ref<bool (swift::CompilerInstance&)>) + 56
30 swift-frontend           0x00000001007171a0 swift::performFrontend(llvm::ArrayRef<char const*>, char const*, void*, swift::FrontendObserver*) + 4292
31 swift-frontend           0x00000001006c4c88 main + 784
32 dyld                     0x000000010ba3d08c start + 520
zsh: abort      swift file.swift
```

</details>

> I have 21 different development snapshots on my Mac, ranging from January 2020 to May 2022. They take up around 75 GB of disk space. I got them from an effort to pinpoint some old AutoDiff bugs, but kept them because they're super useful. That's how I was able to locate the time period so fast.

For 2020-07-22, the diagnostic could just cover up a crash that would happen later on. I hypothesize that the bug originated much earlier than late 2020 - early 2021 (Swift 5.3 - 5.4). Perhaps this situation was never accounted for in the Swift compiler to begin with.

--- a/lib/Sema/CSSimplify.cpp
+++ b/lib/Sema/CSSimplify.cpp
@@ -3140,13 +3140,12 @@ ConstraintSystem::matchFunctionTypes(FunctionType *func1, FunctionType *func2,
       return result;
   }
 
-  if (hasLabelingFailures) {
-    SmallVector<Identifier, 4> correctLabels;
-    for (const auto &param : func2Params)
-      correctLabels.push_back(param.getLabel());
+  if (hasLabelingFailures && !hasFixFor(loc)) {
+    ConstraintFix *fix =
+        loc->isLastElement<LocatorPathElt::ApplyArgToParam>()
+            ? AllowArgumentMismatch::create(*this, func1, func2, loc)
+            : ContextualMismatch::create(*this, func1, func2, loc);
 
-    auto *fix = RelabelArguments::create(*this, correctLabels,
-                                         getConstraintLocator(argumentLocator));
     if (recordFix(fix))
       return getTypeMatchFailure(argumentLocator);
   }

--- a/validation-test/Sema/type_checker_crashers_fixed/issue59058.swift
+++ b/validation-test/Sema/type_checker_crashers_fixed/issue59058.swift
@@ -1,0 +1,38 @@
+// RUN: %target-typecheck-verify-swift
+
+// https://github.com/apple/swift/issues/59058
+
+struct MPSGraphTensor { }
+
+struct MPSGraph {
+  func addition(
+    _ primaryTensor: MPSGraphTensor,
+    _ secondaryTensor: MPSGraphTensor,
+    name: String?
+  ) {
+
+  }
+}
+
+struct Tensor<T> {} // expected-note {{type declared here}}
+
+struct _ExecutionContext {
+  func performBinaryOp<T>(
+    _ lhs: Tensor<T>,
+    _ rhs: Tensor<T>,
+    _ op: (MPSGraphTensor, MPSGraphTensor, String?) -> MPSGraphTensor
+  ) -> Tensor<T> {
+    fatalError()
+  }
+}
+
+public enum _RawTFEager {
+  public static func addV2<T>( // expected-error {{method cannot be declared public because its parameter uses an internal type}}
+    _ x: Tensor<T>
+  ) -> Tensor<T> {
+    return _ExecutionContext.performBinaryOp(x, x, MPSGraph.addition)
+    // expected-error@-1 {{instance member 'performBinaryOp' cannot be used on type '_ExecutionContext'; did you mean to use a value of this type instead?}}
+    // expected-error@-2 {{instance member 'addition' cannot be used on type 'MPSGraph'; did you mean to use a value of this type instead?}}
+    // expected-error@-3 {{cannot convert value of type '(MPSGraphTensor, MPSGraphTensor, String?) -> ()' to expected argument type '(MPSGraphTensor, MPSGraphTensor, String?) -> MPSGraphTensor'}}
+  }
+}

